### PR TITLE
added membership varition dropdown on signup page

### PR DESCRIPTION
**Pull request by rcdtroly**
### Why:
* Allow customer to select membership variation during signup
### What:
* If selected membership has variations, show variation dropdown
* Register the membership with variation if applicable

Alt text

### Refs:
* https://app.asana.com/0/939825968139797/1118021704484376

**NOTE:** this a copy of PR #34 to reduce the conflicting code.

### CODE REVIEW

**Comment by rcdtroly** [comment]
@JamesEmpireOne @atifempireone is this good to go?

File: includes/frontend/views/registration_form.php [hunk 1]
```diff
@@ -217,14 +223,25 @@
                           <?php if($current_membership_id == $membership_option->id){ _e( 'Selected', 'wp99234' );}else{ _e( 'Select', 'wp99234' );}?>
                         </button>
 
+                        <!-- enable membership variations for variation membership only -->
+                        <?php if($membership_option->is_variation_membership): ?>
```
Comment (JamesEmpireOne) [inline includes/frontend/views/registration_form.php]: Where is `is_variation_membership` being set?

Comment (rcdtroly) [reply includes/frontend/views/registration_form.php]: @JamesEmpireOne yes see: Alt text

Comment (rcdtroly) [reply includes/frontend/views/registration_form.php]: @JamesEmpireOne added validation

**Pull request merged**